### ISSUE TITLE
Icon spacing issues

### ISSUE THREAD

**Issue by joshuap** [taffybar/taffybar#318]
The big gap between the "game" and "tmp" workspaces is caused (I think) by the CSS padding on `.IconSpacing`:

![image](https://user-images.githubusercontent.com/474649/39886221-a4373e1e-5443-11e8-88c2-dd0409aa7f43.png)

How to reproduce:

1. Boot a fresh instance of `taffybar`
2. Launch a bunch of `kitty` terminals (or I assume any terminal which doesn't provide an icon) -- you will see the spacing expand for each new window opened.
3. Close all windows
4. Notice that the spacing is not reduced.

**Comment by IvanMalison** [comment]
@joshuap Can you please post the taffybar.hs and taffybar.css that exhibits this behavior?

**Comment by IvanMalison** [comment]
Just in case this ever comes up again, here is the relevant chat log:
```
hah i see,
yeah this log statement is pretty dense
hmm okay so the window state is definitley okay
oh i see
you have .IconImage {
padding-left: 2px;
padding-right: 2px;
}
so i think what you need to do
hmm
yeah so the way the icons work, is that they actually just set their pixbuf to an empty value when there is no window present
so the icon still exists, its just in a dormant state
Ivan Malison @IvanMalison 12:07
there are good reasons for this, but they are kind of complicated to explain
so basically, what we can do is put another class on the IconImage when it is inactive
like i guess .Incative
and then you can use a selector for both of those classes to set the padding to nothing in that case
@joshuap I guess the thing that doesn't make sense to me is that this only happens for you with windows that have no icon
it should happen whenever you add widgets to a workspace and then later remove them
can you verify that that isn't the case?
Joshua Wood @joshuap 12:11
Yeah let me try with something else
Ivan Malison @IvanMalison 12:11
to clarify: this should happen whenever you add some number of widgets to a workspace, and then empty the workspace
Joshua Wood @joshuap 12:12
ah yes, it is the same with all windows
Ivan Malison @IvanMalison 12:12
yeah okay that makes more sense
Joshua Wood @joshuap 12:12
so basically, the icon goes away, but the padding remains
even after the windows are closed
Ivan Malison @IvanMalison 12:12
yeah because whats happening is that the widget is still there
it just has no icon set
Joshua Wood @joshuap 12:13
Makes sense. Are there any memory or performance concerns over not collecting the widgets after windows are removed?
i.e. if I launch 100 windows in a workspace over the span of a few days, then there will be 100 inactive widgets?
Ivan Malison @IvanMalison 12:13
yeah i mean the memory footprint of these widgets is pretty small once the buffer for the actual image is gone
the buffers are recycled
Joshua Wood @joshuap 12:14
OK, cool
Ivan Malison @IvanMalison 12:14
and the reason we do things this way is actually related to performance
Joshua Wood @joshuap 12:14
well an inactive class sounds like it would do the trick
or better yet, maybe an .Active class
Ivan Malison @IvanMalison 12:14
yeah. I'm trying to see if it might already be there
yeah it is, actually
https://github.com/taffybar/taffybar/blob/fd53ba7331e48d9da48b1b5ef9292594a6ff68ff/src/System/Taffybar/Widget/Workspaces.hs#L687
those are the classes that are available on the icon button
which contains the IconImage
so basically, you'll need to use the CSS parent selection
the image is a direct descendant of that object
my CSS is bad so I can't dictate it to you off the top of my head
but its basically already possible
```

**Comment by IvanMalison** [comment]
As I mentioned in the gitter chat, the fix that was put in place in #320 introduces a new issue:

Joshua Wood @joshuap 14:23
@IvanMalison I think I found another small issue with the CSS classes. When I switch between workspaces, the padding on the icons is shifting around slightly. It looks like maybe a .Normal class is added/removed when workspaces are changed?

Ivan Malison @IvanMalison 15:56
@joshuap The issue is going to be that there is a .Active class on the active window icon
basically, i think the best solution is going to be to leave the default styling on .IconImage and then set something like .IconImage .Nodata to have no padding etc.

**Comment by IvanMalison** [comment]
@joshuap Think you can take a swing at fixing this or should I do it?

**Comment by joshuap** [comment]
Oh, I think I got it:

```css
.IconImage {
	padding-left: 2px;
	padding-right: 2px;
}

.nodata .IconImage {
	padding-left: 0px;
	padding-right: 0px;
}
```

It looks like maybe the case in `possibleStatusStrings` is misleading:

https://github.com/taffybar/taffybar/blob/00e61b62212e0e5cb6d3fadf7930d70d96dbcee1/src/System/Taffybar/Widget/Workspaces.hs#L687

I noticed that "nodata" is lowercase here:

https://github.com/taffybar/taffybar/blob/00e61b62212e0e5cb6d3fadf7930d70d96dbcee1/src/System/Taffybar/Widget/Workspaces.hs#L705

If you're OK with this default CSS as a fix for this issue, I'm happy to submit a PR.

**Comment by IvanMalison** [comment]
@joshuap yeah fix the casing and submit a PR
